File: selectionSort.cpp
#include <bits/stdc++.h>
using namespace std;

void selectionSort(int arr[], int n)
{
    int i = 0;
    while (i < n - 1)
    {
        int min_idx = i;
        int j = i + 1;
        while (j < n)
        {
            if (arr[j] < arr[min_idx])
            {
                min_idx = j;
            }
            j++;
        }

        if (min_idx != i)
        {
            swap(arr[min_idx], arr[i]);
        }
        i++;
    }
};

void randomArray(int arr[], int size, int minRange, int maxRange)
{
    cout << "Size : " << endl;
    for (int i = 0; i < size; i++)
    {
        arr[i] = rand() % (maxRange - minRange + 1) + minRange;
    }
}

void printArr(int arr[], int size)
{
    for (int i = 0; i < size; i++)
    {
        cout << arr[i] << " ";
    }

    cout << endl;
}

int main()
{
    int n, minRange, maxRange;
    n = 50000;
    minRange = 0;
    maxRange = 500000;
    
    int *arr = new int[n];

    srand(static_cast<unsigned int>(time(0)));
    randomArray(arr, n, minRange, maxRange);

    cout << "Original array: ";
    printArr(arr, n);

    selectionSort(arr, n);

    cout << "Sorted array: ";
    printArr(arr, n);

    return 0;
}

File: selectionSort.py
import random

def selection_sort(arr):
    n = len(arr)
    for i in range(n-1):
        min_idx = i
        for j in range(i+1, n):
            if arr[j] < arr[min_idx]:
                min_idx = j
        if min_idx != i:
            arr[i], arr[min_idx] = arr[min_idx], arr[i]

def random_array(size, min_range, max_range):
    return [random.randint(min_range, max_range) for _ in range(size)]

def print_arr(arr):
    print(" ".join(map(str, arr)))

def main():
    n = 50000
    min_range = 0
    max_range = 50000

    arr = random_array(n, min_range, max_range)

    print("Original array:", end=" ")
    print_arr(arr)

    selection_sorted_arr = arr.copy()
    selection_sort(selection_sorted_arr)
    print("Sorted array (Selection Sort):", end=" ")
    print_arr(selection_sorted_arr)

if __name__ == "__main__":
    main()
